10_onion_routing.asciidoc

```diff
@@ -642,14 +673,19 @@ No one can tell the difference between filler put there by Alice and filler put
 
 ==== Bob constructs the new onion packet
 
-Bob now copies the onion payload into the onion packet, appends the outer HMAC for chan, re-randomizes the session key with the Elliptic Curve multiplication operation, and appends a fresh version byte.
+Bob now copies the onion payload into the onion packet, appends the outer HMAC for chan, re-randomizes the session key (the same way Alice the sender does) with the Elliptic Curve multiplication operation, and appends a fresh version byte.
 
-////
-TODO
+In order to re-randomize the session key, bob first computes the blinding factor for his hop, using his node public key and the shared secret he derived:
+```
+b_bob = SHA256(P_bob || shared_secret_bob)
+```
 
-Clarify how the HMAC and session key are computed for this onion packet
+With this generated, Bob now re-randomizes the session key by performing an EC multiplication using his session key and the blinding factor:
+```
+session_key_chan = session_key_bob * b_bob
+```
 
-////
+The `session_key_chan` public key will then be appended to the front of the onion packet for processing by Chan>
```

Comment:
There's a typo at the end of this line, I think you meant '.' instead of '>'

Reply:
Yep, slight typo